f8a_report/helpers/report_helper.py

```diff
@@ -93,43 +95,60 @@ def cleanup_feedback(self, table_name, foreign_key, key, ftable_name, column_nam
         logger.debug('Starting to clean up recommendation feedback table')
         # Executing query
         self.cursor.execute(query, (num_days,))
+        # Commiting
         self.conn.commit()
         logger.info('Cleanup of  recommendation_feedback table has completed with status %r',
                     self.cursor.statusmessage)
 
     def cleanup_db_tables(self):
         """Cleanup RDS data tables on a periodic basis."""
-        try:
+        def clean_celery_taskmeta():
             # Number of days to retain the celery task_meta data
             num_days = os.environ.get('KEEP_DB_META_NUM_DAYS', '30')
             self.cleanup_tables('celery_taskmeta', 'date_done', num_days)
 
+        def clean_worker_results():
             # Number of days to retain the worker results data
             num_days = os.environ.get('KEEP_WORKER_RESULT_NUM_DAYS', '30')
             self.cleanup_tables('worker_results', 'ended_at', num_days)
 
+        def clean_recommendation_feedback():
             # Number of days to retain recommendation feedback data data
             num_days = os.environ.get('KEEP_RECOMMENDATION_FEEDBACK_NUM_DAYS', '180')
             self.cleanup_feedback('recommendation_feedback', 'stack_id', 'id',
                                   'stack_analyses_request', 'submitTime', num_days)
 
+        def clean_stack_analyses_request():
             # Number of days to retain the stack analyses request data
             num_days = os.environ.get('KEEP_STACK_ANALYSES_REQUESTS_NUM_DAYS', '181')
             self.cleanup_tables('stack_analyses_request', 'submitTime', num_days)
 
+        def clean_api_requests():
             # Number of days to retain the api requests data
             num_days = os.environ.get('KEEP_API_REQUESTS_NUM_DAYS', '180')
             self.cleanup_tables('api_requests', 'submit_time', num_days)
 
+        def clean_package_worker_result():
             # Number of days to retain the package worker results data
             num_days = os.environ.get('KEEP_PACKAGE_WORKER_RESULT_NUM_DAYS', '30')
             self.cleanup_tables('package_worker_results', 'ended_at', num_days)
 
+        def clean_package_analyses():
             # Number of days to retain the package analyses data
             num_days = os.environ.get('KEEP_PACKAGE_ANALYSES_NUM_DAYS', '31')
             self.cleanup_tables('package_analyses', 'finished_at', num_days)
-        except Exception as e:
-            logger.error('CleanupDatabaseError: %r' % e)
+
+        # Storing all the cleanup function objects in a variable
+        clean_tables = [clean_celery_taskmeta, clean_worker_results, clean_recommendation_feedback,
```

Comment:
Is this really required, if you catch the exception, log it and dont throw it, can you not keep executing the code.

Reply:
I did not find something like that in python.where we can just catch an exception.

Reply:
try:
    <code>
except:
    log('something')
print('anything')

It will print 'anything'. The exception has been handled and we have moved on.

Reply:
doing this individually for every table clean up increases the maintainability index of the file.

Reply:
why should it be done individually, it can be handled in the cleanup_tables. Catch the exception, log it and return.

Reply:
Okay.

Reply:
If you handle cleanup_tables level, single exception can block rest of the table cleanup. Is that okay?

Reply:
I think @prashbnair is suggesting it wont block rest of the table cleanup.

Reply:
I am trying it locally.

Reply:
It seems to work.I had entered a wrong column name in the query as a test and after that failed the other cleanup continued.